R/drive_trash.R

```diff
@@ -0,0 +1,111 @@
+#' Trash file on Google Drive.
+#'
+#' @template file
+#' @template verbose
+#'
+#' @return Logical, indicating whether the trashing succeeded.
+#' @export
+#' @examples
+#' \dontrun{
+#' ## Trash a single file in the trash.
+#' drive_trash("chickwts.csv")
+#'
+#' ## Trash multiple files in trash.
+#' drive_trash(c("abc", "def"))
+#' }
+drive_trash <- function(file = NULL, verbose = TRUE) {
+  trash_file <- as_dribble(file)
+  if (!some_files(trash_file) && verbose) {
+    message(glue("No such files found to trash"))
+  }
+
+  out <- purrr::map_lgl(trash_file$id, trash_one)
+
+  if (verbose) {
+    if (any(out)) {
+      successes <- glue_data(trash_file[out, ], "  * {name}: {id}")
+      message(collapse(c("Files trashed:", successes), sep = "\n"))
+    }
+    if (any(!out)) {
+      failures <- glue_data(trash_file[!out, ], "  * {name}: {id}")
+      message(collapse(c("Files NOT trashed:", failures), sep = "\n"))
+    }
+  }
+  invisible(out)
+}
+
+trash_one <- function(id, trash = TRUE) {
+  request <- generate_request(
+    endpoint = "drive.files.update",
+    params = list(fileId = id,
+                  trashed = trash)
+  )
+  proc_res <- do_request(request, encode = "json")
+  identical(proc_res$id, id)
+}
+
+#' Untrash file on Google Drive.
```

Comment:
These could conceivably be documented together. That would make the examples nice.